### ISSUE TITLE
possible resource leak in IOUtils.readRawJsonArray()

### ISSUE THREAD

**Issue by mTwTm** [mozilla-tw/FirefoxLite#3162]
```
public static JSONArray readRawJsonArray(Context context, int resId) throws IOException {
        try {
            InputStream is = context.getResources().openRawResource(resId);
            int size = is.available();
            byte[] buffer = new byte[size];
            is.read(buffer);
            String json = new String(buffer, "UTF-8");
            is.close();
            return new JSONArray(json);
        } catch (JSONException e) {
            throw new AssertionError("Corrupt JSON in readRawJsonArray", e);
        }
    }
```

We might want to fix it with autoclosable.

**Comment by josephyu1110** [comment]
[TopSitesUtils.loadDefaultSitesFromAssets](https://github.com/mozilla-tw/FirefoxLite/blob/05c0cfd77ae66bf686a8c1577aebdc7b58c92a87/app/src/main/java/org/mozilla/focus/utils/TopSitesUtils.java#L52-L66) may have the same risk.

**Comment by mTwTm** [comment]
Thanks! I'll file a separate bug for it. (Don't think it's critical for now so I'll leave them there and label good first issue.)

**Comment by mTwTm** [comment]
Hmm, maybe we can leave them in the same bug. The link is already self-explanatory so I'll not quote the code here.

**Comment by pengrad** [comment]
@mTwTm  Is it better to use "Closes #..." commit message to automatically close issue?
I just checked top commit history and saw "Issue #..." pattern :)

**Comment by mTwTm** [comment]
@pengrad Personally I'm using neither but @cnevinc is thinking of standardizing that.
@pengrad can you help open an issue with the topic `Standardize commit messages` so that we can update the progress there?

**Comment by mTwTm** [comment]
Fixed in #3497
